src/controllers/auth.ts

```diff
@@ -50,4 +53,48 @@ const login = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export default login;
+const register = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { body } = req;
+    await userRegisterSchema.validate(body);
+    if (body.role === 'therapist') {
+      const user = await registerTherapist(body);
+      const mailGenerator = new Mailgen({
+        theme: 'salted',
+        product: {
+          name: 'Ntherapy',
+          link: 'https://ntherapy.com/',
+        },
+      });
+      const email = {
+        body: {
+          name: user.fullName,
+          intro: 'Welcome to Ntherapy! We\'re very excited to have you on board. \n we will review your application and get back to you as soon as possible.',
+          outro: 'Need help, or have questions? Just reply to this email, we\'d love to help.',
+        },
+      };
+      const emailBody = mailGenerator.generate(email);
+      const emailText = mailGenerator.generatePlaintext(email);
+      await mailer({
+        to: user.email,
+        subject: 'Account Activation',
+        text: emailText,
```

Comment:
Not sure if both text and html are required, I don't assume that.
@ahmedmurtaja